modules/statistics/php/stats_demographic.class.inc

```diff
@@ -163,203 +165,214 @@ class Stats_Demographic extends \NDB_Form
         foreach (\Utility::getProjectList() as $key => $value) {
             $projects[$key] = $value;
         }
-        $currentProject = null;
-        if ($_REQUEST['DemographicProject'] ?? ''
-        ) {
-            $currentProject = htmlspecialchars($_REQUEST['DemographicProject']);
+
+        $demographicProject = $_REQUEST['DemographicProject'] ?? '';
+        $paramProject       = '';
+
+        if (!empty($demographicProject)) {
+            $demographicProject = htmlspecialchars($demographicProject);
             $this->tpl_data['CurrentProject']
                 = array(
-                   'ID'   => $currentProject,
-                   'Name' => $projects[$currentProject],
+                   'ID'   => $demographicProject,
+                   'Name' => $projects[$demographicProject] ?? '',
                   );
-            $Param_Project       ='AND (c.ProjectID IS NULL OR c.ProjectID=:pid) ';
-            $this->params['pid'] = htmlspecialchars($_REQUEST['DemographicProject']);
-        } else {
-            $Param_Project ='';
+            $paramProject        ='AND (c.ProjectID IS NULL OR c.ProjectID=:pid) ';
+            $this->params['pid'] = $demographicProject;
         }
 
         //SUBPROJECTS
-        $subprojList ="";
-        $subprojects = \Utility::getSubprojectsForProject($currentProject);
-        foreach ($subprojects as $key=>$val) {
-            $subprojList .= $key.",";
-        }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
-        } else {
-            $suproject_query ='';
+        $subprojList     = '';
+        $subprojectQuery = '';
+        $visits          = \Utility::getExistingVisitLabels($demographicProject);
+        $project         = $demographicProject != '' ? null : $demographicProject;
```

Comment:
what's this line doing?

Reply:
@johnsaigle this isn't really the same as `$projectID = empty($projectID) ? null : $projectID` .. it's actually reading `$projectID = !empty($projectID) ? null : $projectID`which doesn't really make sense to me

Reply:
Good catch!!! Thanks